From a62eea47e38313ebabe787d4b972b0bbb3af9ada Mon Sep 17 00:00:00 2001
From: Sylvain Munaut <tnt@246tNt.com>
Date: Tue, 25 Aug 2015 09:53:29 -0400
Subject: [PATCH 05/17] cmake: Fix warning related to GrSetupQt4 in modern
 CMake

Corresponding commit in the cmake git tree :
2509c7678feb2efab79a384fd0b0b35cee8d02b0

Signed-off-by: Sylvain Munaut <tnt@246tNt.com>
---
 cmake/Modules/GrSetupQt4.cmake | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/cmake/Modules/GrSetupQt4.cmake b/cmake/Modules/GrSetupQt4.cmake
index 312b415..cc85c39 100644
--- a/cmake/Modules/GrSetupQt4.cmake
+++ b/cmake/Modules/GrSetupQt4.cmake
@@ -67,13 +67,7 @@ set(__INCLUDED_GR_USEQT4_CMAKE TRUE)
 ################################################################
 
 ADD_DEFINITIONS(${QT_DEFINITIONS})
-SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS_DEBUG QT_DEBUG)
-SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS_RELEASE QT_NO_DEBUG)
-SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS_RELWITHDEBINFO QT_NO_DEBUG)
-SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS_MINSIZEREL QT_NO_DEBUG)
-IF(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
-  SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS QT_NO_DEBUG)
-ENDIF()
+SET_PROPERTY(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG>)
 
 SET(QT_INCLUDE_DIRS ${QT_INCLUDE_DIR})
 SET(QT_LIBRARIES "")
-- 
2.1.4

